pkg/releaser/releaser.go

```diff
@@ -107,35 +105,22 @@ func (r *Releaser) UpdateIndexFile() (bool, error) {
 		}
 	}
 
-	var indexFile *repo.IndexFile
-
-	resp, err := r.httpClient.Get(fmt.Sprintf("%s/index.yaml", r.config.ChartsRepo))
+	fmt.Printf("Loading index file from git repository %s\n", r.config.IndexPath)
+	worktree, err := r.git.AddWorktree("", r.config.Remote+"/"+r.config.PagesBranch)
 	if err != nil {
 		return false, err
 	}
+	defer r.git.RemoveWorktree("", worktree) // nolint: errcheck
+	indexYamlPath := filepath.Join(worktree, "index.yaml")
 
-	defer resp.Body.Close()
-
-	if resp.StatusCode == http.StatusOK {
-		out, err := os.Create(r.config.IndexPath)
-		if err != nil {
-			return false, err
-		}
-		defer out.Close()
-
-		_, err = io.Copy(out, resp.Body)
-		if err != nil {
-			return false, err
-		}
-
-		fmt.Printf("Using existing index at %s\n", r.config.IndexPath)
-		indexFile, err = repo.LoadIndexFile(r.config.IndexPath)
+	var indexFile *repo.IndexFile
+	if _, err := os.Stat(indexYamlPath); errors.Is(err, os.ErrNotExist) {
```

Comment:
In case of another error, the error should be returned here.

Reply:
thanks for spotting this.